tweets/views.py

```diff
@@ -1,9 +1,44 @@
-from django.contrib.auth.mixins import LoginRequiredMixin
-from django.views.generic import TemplateView
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.urls import reverse_lazy
+from django.views.generic import CreateView, DeleteView, DetailView, ListView
 
+from .forms import TweetForm
+from .models import Tweet
 
-class HomeView(LoginRequiredMixin, TemplateView):
+
+class HomeView(LoginRequiredMixin, ListView):
+    model = Tweet
     template_name = "tweets/home.html"
+    ordering = "-created_at"
+    queryset = model.objects.select_related("user")
+
+
+class TweetCreateView(LoginRequiredMixin, CreateView):
+    template_name = "tweets/create.html"
+    model = Tweet
+    form_class = TweetForm
+    success_url = reverse_lazy("tweets:home")
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        return super().form_valid(form)
+
+
+class TweetDetailView(LoginRequiredMixin, DetailView):
+    template_name = "tweets/detail.html"
+    model = Tweet
+    queryset = Tweet.objects.select_related("user")

```

Comment:
細かいですが、ここは
```suggestion
    queryset = model.objects.select_related("user")
```
とすると統一感が合ってよいと思います。